platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DynamicCallProblemReporting.kt

```diff
@@ -66,17 +68,38 @@ class DefaultDynamicCallProblemReporting : DynamicCallProblemReporting {
     }
 
     override fun leaveDynamicCall(entryPoint: Any) {
+        if (currentThreadState.callStack.isEmpty()) {
+            return
+        }
         val innermostCall = currentThreadState.callStack.pop()
         check(entryPoint == innermostCall.entryPoint) { "Mismatched enter-leave calls in DynamicCallProjectIsolationProblemReporting" }
     }
 
     override fun unreportedProblemInCurrentCall(problemKey: Any): Boolean {
         val currentThreadCallStack = currentThreadState.callStack
-        check(currentThreadCallStack.isNotEmpty()) { "Expected unreportedProblemInCurrentCall to be called after enterDynamicCall" }
+        if (currentThreadCallStack.isEmpty()) {
+            logMissingScope()
+            return true
+        }
```

Comment:
This is the main change. Instead of failing when `enterDynamicCall` was not called we consider a problem unreported.

This has a side effect of that problems might get "duplicated" until the dynamic call scope is actually installed with a call to `enterDynamicCall`. An example of the duplication is a property lookup violation that is repeated for each transitive parent of a project doing the lookup. 

We relax the requirement for now to unblock builds suffering from this. In the future, we should find a solution that does not duplicate the problems.